Support multiple CSV files in series

cottsay opened the pull request:
Since the CSVSeries supports multiple data points anyway, allow a wildcard pattern to match multiple files and add them all to the same series.

<!-- Link to Jira ticket, https://issues.jenkins-ci.org
     good to have, but optional
Jira: [JENKINS-XXXXX](https://issues.jenkins-ci.org/browse/JENKINS-XXXXX)
 -->

<!-- Comment:
If the issue is not fully described in the ticket, add more information here (justification, pull request links, etc.).

 * JIRA issues for minor improvements are not mandatory.
 * Bugfixes should have a JIRA issue (backporting process).
 * Major new features should have a JIRA issue reference.
 * Issue ID should be included in PR name (e.g. "[JENKINS-XXXXX] Add feature X.Y.Z".

### What has been done
1.
2.
-->

<!-- optional. Should be removed if not applicable
### Screenshots

Before | After
:-: | :-:
| |
-->

### How to test
Specify a pattern to a CSV file which matches multiple files.
Note that a test was added to verify this.

### Checklist

- [x] Git commits follow [best practices](https://chris.beams.io/posts/git-commit/) <!-- mandatory -->
- [ ] Build passes in Jenkins <!-- mandatory -->
- [x] Appropriate tests or explanation to why this change has no tests <!-- mandatory -->
- [ ] JIRA issue is well described (problem explanation, steps to reproduce, screenshots) <!-- optional -->
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

vgaidarji commented on src/main/java/hudson/plugins/plot/CSVSeries.java:
```diff
@@ -109,36 +109,55 @@ public boolean getDisplayTableFlag() {
+            return null;
+        }
+
+        for (int index = 0; index < seriesFiles.length; index++) {
```
how about `foreach` syntax instead?
```suggestion
        for (FilePath seriesFile : seriesFiles) {
            List<PlotPoint> seriesList = loadSeriesFile(seriesFile, buildNumber, logger);
```

cottsay commented on src/test/java/hudson/plugins/plot/CSVSeriesTest.java:
```diff
@@ -154,4 +154,20 @@ private int getNumColumns(FilePath workspaceRootDir, String file) throws IOExcep
+            // load the series.
+            List<PlotPoint> points = series.loadSeries(workspaceRootDir, 0, System.out);
+            LOGGER.info("Got " + points.size() + " plot points");
+            testPlotPoints(points, 2);  // Avg is found in 2 files, each containing 1 line
```
I created some separate test collateral for this case, and used the existing tests to verify functionality. Should be a lot less fragile.

cottsay commented:
Thanks very much for the swift and helpful feedback, @vgaidarji. I believe I have addressed all of your concerns.

Please squash these commits prior to merging - I left them fanned out to facilitate review.

vgaidarji commented:
@cottsay thank you. I'll merge and release the new version soon 👍

vgaidarji commented:
The new version of the plugin [2.1.3](https://github.com/jenkinsci/plot-plugin/releases/tag/plot-2.1.3) was released and should be available in [update center](https://updates.jenkins-ci.org/download/plugins/plot/) soon. But it was already uploaded to repo https://repo.jenkins-ci.org/releases/org/jenkins-ci/plugins/plot/2.1.3/.
I'll need to migrate release notes from old wiki format to GitHub and will include release notes for 2.1.3 there.

@cottsay, thank you very much for your contribution 🤝🎉
